src/rgw/rgw_rados.cc

```diff
@@ -1503,7 +1503,7 @@ int RGWRados::init_complete()
     run_sync_thread = false;
   }
 
-  if (run_sync_thread) {
+  if (svc.zone->need_to_log_metadata()) {
```

Comment:
apologies @adamemerson, i was mistaken about this one. `need_to_log_metadata()` only returns true on the metadata master zone, but other zones need this too for mdlog trimming

i think `!current_period.get_id().empty()` would be succifient

Reply:
@cbodley Fixed!